lib/ddtrace/contrib/mongodb/instrumentation.rb

```diff
@@ -51,7 +51,7 @@ def self.included(base)
           module InstanceMethods
             def datadog_pin
               @datadog_pin ||= begin
-                service = Datadog.configuration[:mongo][:service_name]
+                service = Datadog.configuration[:mongo, self.to_param][:service_name]
```

Comment:
It seems like [`Mongo::Address#to_param` is not a valid method](https://app.circleci.com/pipelines/github/DataDog/dd-trace-rb/3044/workflows/e2e5682d-8d15-4224-b956-ec7b98666972/jobs/121786#failed-test-476).
We could probably use `Mongo::Address#seed` here to get the same kind of output.

Reply:
Yes, you are right... somehow it worked on my local with ruby 2.7 - I changed it to seed.
Anyway, I'll try to add specs for two mongo clients as you suggested soon.
Right now, unfortunately, I cannot get the CI test run green even with existing specs...

Reply:
Apologies for the CI brokenness @skcc321. I've raised this internally, we'll look into this ASAP as it's breaking for every PR. I'll leave a quick note here once it's fixed.

Reply:
😓  a few more things broke due to upstream releases, we're still working to get CI back to green.

Reply:
CI is now back to green 🎉  . A rebase on top of current `master` should get you a nice starting state :)

Reply:
still some failures. But anyway, testing is possible now